src/gui/guiFormSpecMenu.h

```diff
@@ -22,6 +22,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include <utility>
 #include <stack>
 
+#include "client/keycode.h"
```

Comment:
Only needed in `guiFormSpecMenu.cpp`. Here you can use the forward declaration `class KeyPress;` for compiling speed reasons.

Reply:
No idea how that works.  C++ is too weird and bloated.  Do I place `class KeyPress` anywhere in the file, or as part of the variable type?

Reply:
Afaik, all you have to do is to write the class header:
```cpp
// include everything

class KeyPress;

// declare other class that has a member variable that needs the class KeyPress
```
It's just like the function forward declaration which is done in header files.

Reply:
You can replace this commented line with `class KeyPress;`. Then undo the removal of `#include "client/keycode.h"` in the cpp file.